### ISSUE TITLE
Seg fault while post quantizing my model

### ISSUE THREAD

**Issue by shreymohan** [tensorflow/tensorflow#40359]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): OS Linux 18.04.04 LTS
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): python wheel
- TensorFlow version (use command below): 2.2.0-rc4
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:10.1
- GPU model and memory: TitanV , 12 GB

I am trying to do post training quantization for the EfficientDet-D0 model pre-trained on COCO I got from his repo : https://github.com/google/automl  
Desired quantization is int8 because it needs to be running on a google coral.

Below is the code that I use:

`from PIL import Image
import numpy as np
export_dir = "./savedmodeldir"

model = tf.saved_model.load(export_dir)

concrete_func = model.signatures[tf.saved_model.DEFAULT_SERVING_SIGNATURE_DEF_KEY]
concrete_func.inputs[0].set_shape([1, 512, 512, 3])
converter = tf.lite.TFLiteConverter.from_concrete_functions([concrete_func])

converter.optimizations = [tf.lite.Optimize.DEFAULT]
def representative_dataset_gen():
  num_calibration_steps=10
  path_to_img='./img/img.jpg'
  for _ in range(num_calibration_steps):
    # Get sample input data as a numpy array in a method of your choosing.
    im = np.asarray(Image.open(path_to_img).resize((512,512)),dtype=np.uint8)
    im=np.reshape(im, [1,512,512,3])
    with tf.device('/CPU:0'):
      image=tf.convert_to_tensor(im)
    yield [image]
converter.representative_dataset = representative_dataset_gen
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]
converter.inference_input_type = tf.int8  # or tf.uint8
converter.inference_output_type = tf.int8  # or tf.uint8

tflite_quant_model = converter.convert()`

Link to the savedmodeldir zip file:
https://drive.google.com/file/d/1Hsbox0LRm6z9FecnobMP7YK7ZJyFjknt/view?usp=sharing

The output I get while running the code:

WARNING:tensorflow:Importing a function (MetaGraph import) with ops with custom gradients. Will likely fail if a gradient is requested.
WARNING:tensorflow:Importing a function (MetaGraph import) with ops with custom gradients. Will likely fail if a gradient is requested.
WARNING:tensorflow:From /home/shrey/.local/lib/python3.6/site-packages/tensorflow/python/ops/resource_variable_ops.py:1817: calling BaseResourceVariable.__init__ (from tensorflow.python.ops.resource_variable_ops) with constraint is deprecated and will be removed in a future version.
Instructions for updating:
If using Keras pass *_constraint arguments to layers.
WARNING:tensorflow:Issue encountered when serializing global_step.
Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.
to_proto not supported in EAGER mode.
WARNING:tensorflow:Issue encountered when serializing moving_average_variables.
Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.
to_proto not supported in EAGER mode.
WARNING:tensorflow:Issue encountered when serializing variables.
Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.
to_proto not supported in EAGER mode.
WARNING:tensorflow:Issue encountered when serializing trainable_variables.
Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.
to_proto not supported in EAGER mode.
2020-06-10 11:40:38.281380: I tensorflow/core/grappler/devices.cc:55] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0
2020-06-10 11:40:38.281495: I tensorflow/core/grappler/clusters/single_machine.cc:356] Starting new session
2020-06-10 11:40:38.432064: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:797] Optimization results for grappler item: graph_to_optimize
2020-06-10 11:40:38.432098: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: Graph size after: 3786 nodes (0), 3573 edges (0), time = 21.187ms.
2020-06-10 11:40:38.432102: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: Graph size after: 3786 nodes (0), 3573 edges (0), time = 26.002ms.
2020-06-10 11:40:38.432105: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:797] Optimization results for grappler item: __inference_TensorArrayV2Write_cond_true_136_1822
2020-06-10 11:40:38.432109: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0.001ms.
2020-06-10 11:40:38.432112: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0ms.
2020-06-10 11:40:38.432115: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:797] Optimization results for grappler item: __inference_cond_png_true_83_1595
2020-06-10 11:40:38.432118: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0ms.
2020-06-10 11:40:38.432121: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0ms.
2020-06-10 11:40:38.432124: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:797] Optimization results for grappler item: __inference_while_body_23_4372
2020-06-10 11:40:38.432128: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: Graph size after: 38 nodes (0), 41 edges (0), time = 0.721ms.
2020-06-10 11:40:38.432131: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: Graph size after: 38 nodes (0), 41 edges (0), time = 0.751ms.
2020-06-10 11:40:38.432134: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:797] Optimization results for grappler item: __inference_TensorArrayV2Write_cond_false_137_1473
2020-06-10 11:40:38.432137: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0ms.
2020-06-10 11:40:38.432140: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0.001ms.
2020-06-10 11:40:38.432143: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:797] Optimization results for grappler item: __inference_cond_gif_true_93_2677
2020-06-10 11:40:38.432148: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0.001ms.
2020-06-10 11:40:38.432151: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0ms.
2020-06-10 11:40:38.432154: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:797] Optimization results for grappler item: __inference_cond_gif_false_94_4320
2020-06-10 11:40:38.432158: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0.001ms.
2020-06-10 11:40:38.432161: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0ms.
2020-06-10 11:40:38.432185: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:797] Optimization results for grappler item: __inference_decode_image_cond_jpeg_true_65_2555
2020-06-10 11:40:38.432190: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0ms.
2020-06-10 11:40:38.432194: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0.001ms.
2020-06-10 11:40:38.432198: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:797] Optimization results for grappler item: __inference_decode_image_cond_jpeg_false_66_4339
2020-06-10 11:40:38.432202: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: Graph size after: 10 nodes (0), 10 edges (0), time = 0.32ms.
2020-06-10 11:40:38.432210: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: Graph size after: 10 nodes (0), 10 edges (0), time = 0.316ms.
2020-06-10 11:40:38.432213: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:797] Optimization results for grappler item: __inference_while_cond_22_6546
2020-06-10 11:40:38.432219: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0ms.
2020-06-10 11:40:38.432222: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: function_optimizer did nothing. time = 0ms.
2020-06-10 11:40:38.432227: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:797] Optimization results for grappler item: __inference_cond_png_false_84_4328
2020-06-10 11:40:38.432230: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: Graph size after: 7 nodes (0), 6 edges (0), time = 0.234ms.
2020-06-10 11:40:38.432234: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   function_optimizer: Graph size after: 7 nodes (0), 6 edges (0), time = 0.248ms.
2020-06-10 11:40:40.776624: I tensorflow/core/grappler/devices.cc:55] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0
2020-06-10 11:40:40.776752: I tensorflow/core/grappler/clusters/single_machine.cc:356] Starting new session
2020-06-10 11:40:41.366677: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:797] Optimization results for grappler item: graph_to_optimize
2020-06-10 11:40:41.366709: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   constant_folding: Graph size after: 2739 nodes (-1047), 2330 edges (-1243), time = 481.837ms.
2020-06-10 11:40:41.366713: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:799]   constant_folding: Graph size after: 2739 nodes (0), 2330 edges (0), time = 49.056ms.
Aborted (core dumped)

**Comment by amahendrakar** [comment]
@shreymohan,
On running the code I am facing a different error stating `error: 'tfl.strided_slice' op operand #0 must be tensor of 32-bit float or 32-bit integer or 64-bit integer or 8-bit integer or QI8 type or QUI8 type or 1-bit integer values, but got 'tensor<1x512x512x3x!tf.quint8>'`. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/1ca7851ea1332eac3c8d3142a0fa4842/40359.ipynb). 

Could you please check if you are facing the same error in a virtual environment. Thanks!

**Comment by shreymohan** [comment]
This is weird. Would you suggest trying tf-nightly instead? I have seen people pulling off their conversions using that tf version.

**Comment by amahendrakar** [comment]
Was able to reproduce the issue, runtime crashes on running the code with TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/89e524b36c99b36ae2157c55fc81aab0/40359-tf-nightly.ipynb). Thanks!

**Comment by shreymohan** [comment]
@amahendrakar  Hey! thanks for reaching out! Any suggestions on how to solve this issue?

**Comment by sanatmpa1** [comment]
@shreymohan,

I tried reproducing the code in TF 2.6.0 and it was throwing RuntimeError  `Failed to quantize: <unknown>:0: error: loc("strided_slice_21"): 'tfl.strided_slice' op quantization parameters violate the same scale constraint: !quant.uniform<i8:f32, 0.81405168771743774:-128> vs. !quant.uniform<i8:f32, 1.4241064786911011:-128>`, whereas in tf-nightly i.e `2.8.0-dev20211020`, it ran successfully. please take a look at the [gist here](https://colab.research.google.com/gist/sanatmpa1/30338cfaba1ab7fb200dccc5b5ac51fd/40359-tf-nightly.ipynb)
